server/conn.go

```diff
@@ -2394,61 +2399,62 @@ func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool
 // binary specifies the way to dump data. It throws any error while dumping data.
 // serverStatus, a flag bit represents server information.
 // fetchSize, the desired number of rows to be fetched each time when client uses cursor.
-func (cc *clientConn) writeChunksWithFetchSize(ctx context.Context, rs ResultSet, serverStatus uint16, fetchSize int) error {
-	fetchedRows := rs.GetFetchedRows()
+func (cc *clientConn) writeChunksWithFetchSize(ctx context.Context, rs cursorResultSet, serverStatus uint16, fetchSize int) error {
+	iter := rs.GetRowIterator()
+
+	// construct the rows sent to the client according to fetchSize.
+	var curRows []chunk.Row
+	for i := 0; i < fetchSize && iter.Current() != iter.End(); i++ {
```

Comment:
This performance is toooooooo bad (because of the redundant chunk allocation). I'm optimizing it right now.